client/src/utils/chooserModal.js

```diff
@@ -0,0 +1,143 @@
+import $ from 'jquery';
+import { gettext } from './gettext';
+
+const submitCreationForm = (modal, form, errorContainerSelector) => {
+  const formdata = new FormData(form);
+
+  $.ajax({
+    url: form.action,
+    data: formdata,
+    processData: false,
+    contentType: false,
+    type: 'POST',
+    dataType: 'text',
+    success: modal.loadResponseText,
+    error(response, textStatus, errorThrown) {
+      const message =
+        gettext(
+          'Report this error to your website administrator with the following information:',
+        ) +
+        '<br />' +
+        errorThrown +
+        ' - ' +
+        response.status;
+      $(errorContainerSelector, modal.body).append(
+        '<div class="help-block help-critical">' +
+          '<strong>' +
+          gettext('Server Error') +
+          ': </strong>' +
+          message +
+          '</div>',
+      );
+    },
+  });
+};
+
+const initPrefillTitleFromFilename = (
+  modal,
+  fileFieldSelector,
+  titleFieldSelector,
+  eventName,
```

Comment:
any chance we can opt for an object for the last three params here, it gives us more freedom to add things later and self-documents when calling the function.

```suggestion
 { fileFieldSelector,
  titleFieldSelector,
  eventName,} = {}
```

So the first param is `modal` and the second param is an `options` object.

Reply:
Good idea! Opts dict added in latest rebase.